October/MetaGen/src/pf_inference.jl

```diff
@@ -1,4 +1,21 @@
-#array for each frame and array for each receptive field and array for those detections
+"""
+    unfold_particle_filter(num_particles::Int, objects_observed::Matrix{Array{Array{Detection2D}}}, camera_trajectories::Matrix{Camera_Params}, num_receptive_fields::Int64)
+
+Performs inference procedure.
+
+# Arguments
+- num_particles::Int
+- objects_observed: *how are these indexed?*
+- camera_trajectories: *how are these indexed?*
```

Comment:
The one thing I'm still not sure about is the indexing on `objects_observed, camera_trajectories` and `num_receptive_fields`. Could you clarify?

Reply:
Yep! So objects_observed and camera_trajectories are both matrices, where each row is for a video, and each column is for a frame. So camera_trajectories[v, f] will contain the Camera_Params for video number v, frame number f. Each element of objects_observed contains an array with the observations indexed by receptive field. So objects_observed[v, f] will be an array of length num_receptive_fields. Indexing by receptive field number, call it r, we have that objects_observed[v, f][r] contains a list of the Detections2D observed in that receptive field.

Reply:
Thank you! If everything else looks good to you, then the PR is ready to merge.